tests/integration/conftest.py

```diff
@@ -11,3 +23,78 @@ def charm():
     # juju bundle files expect local charms to begin with `./` or `/` to distinguish them from
     # Charmhub charms.
     return f"./postgresql-k8s_ubuntu@22.04-{architecture.architecture}.charm"
+
+
+def get_cloud_config(cloud: str) -> tuple[dict[str, str], dict[str, str]]:
+    # Define some configurations and credentials.
+    if cloud == AWS:
+        return {
+            "endpoint": "https://s3.amazonaws.com",
+            "bucket": "data-charms-testing",
+            "path": f"/postgresql-k8s/{uuid.uuid1()}",
+            "region": "us-east-1",
+        }, {
+            "access-key": os.environ["AWS_ACCESS_KEY"],
+            "secret-key": os.environ["AWS_SECRET_KEY"],
+        }
+    elif cloud == GCP:
+        return {
+            "endpoint": "https://storage.googleapis.com",
+            "bucket": "data-charms-testing",
+            "path": f"/postgresql-k8s/{uuid.uuid1()}",
+            "region": "",
+        }, {
+            "access-key": os.environ["GCP_ACCESS_KEY"],
+            "secret-key": os.environ["GCP_SECRET_KEY"],
+        }
+
+
+def cleanup_cloud(config: dict[str, str], credentials: dict[str, str]) -> None:
+    # Delete the previously created objects.
+    logger.info("deleting the previously created backups")
+    session = boto3.session.Session(
+        aws_access_key_id=credentials["access-key"],
+        aws_secret_access_key=credentials["secret-key"],
+        region_name=config["region"],
+    )
+    s3 = session.resource(
+        "s3", endpoint_url=construct_endpoint(config["endpoint"], config["region"])
+    )
+    bucket = s3.Bucket(config["bucket"])
+    # GCS doesn't support batch delete operation, so delete the objects one by one.
+    for bucket_object in bucket.objects.filter(Prefix=config["path"].lstrip("/")):
+        bucket_object.delete()
+
+
+@pytest.fixture(scope="module")
+async def aws_cloud_configs(ops_test: OpsTest) -> None:
```

Comment:
Split the fixture to per cloud. So that we can skip only for the cloud that has missing credentials.